src/main/java/duke/system/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke.system;
+
+import duke.system.exception.DukeException;
+import duke.task.*;
+
+/**
+ * Represents a parser that takes in the entered <code>command</code> by the user and filtered by the enum,
+ * then return the parsed <code>command</code>, <code>argument</code> and <code>date</code>
+ * <code>print</code> the output and add the parsed item to the list if needed
+ */
+public class Parser {
+    private final String command;
+    private final String argument;
```

Comment:
For this variable, I would suggest to give it a Javadoc comment because it is not very obvious what is it at first glance :)